developer_guides/virtualisation/virtualisation.rst

```diff
@@ -0,0 +1,349 @@
+components from its topology, it will be forwarded to the DSP. That IPC contains
+the volume component ID as seen by the respective guest. Therefore it should be
+exactly the same ID, as the one used by the firmware.
+
+2.3. DPCM audio routing
+-----------------------
+
+In the most trivial case when a user-space application opens an audio interface,
+there is a unique sequence of audio components, that will comprise a DPCM audio
+route, involved in this operation. Such a sequence is called an audio pipeline.
+However, modern audio hardware often contains more complex audio graphs, also
+sometimes requiring dynamic re-routing. Support for such configurations is
+provided by the ASoC DPCM_ API. We also use this API to activate and deactivate
+guest audio interfaces.
+
+3. Implementation
+*****************
+
+3.1. Initialisation
+-------------------
+
+The SOF driver probing on the host remains unmodified, except that additionally
+the vhost driver is also initialised, which registers a newly added
+/dev/vhost-dsp misc device, used by guests to establish VirtQueue links and
+control the SOF vhost driver, using dedicated vhost ioctls.
+
+When QEMU starts a guest instance, it detects the above misc device and
+creates a PCI device for the guest with dedicated vendor and device IDs. All
+vhost PCI devices use the Qumranet / RedHat vendor ID and are allocated a
+device ID from a range, specially donated by RedHat for this purpose.
+
+The SOF VirtIO driver on the guest registers support for that PCI device and
+its .probe() method is called.
+
+During the SOF VirtIO driver probing, the QEMU SOF code reports to the vhost
+driver, which VM image is being instantiated. This is used as a ewliable guest
+identification and is needed for SOF guest audio topology selection.
+
+The SOF driver on the guest performs partial initialisation, it skips any
+steps, involving communicating with the actual DSP, e.g. sending the firmware,
+booting it, initialisation the tracing interface.
+
+The VirtIO driver on the guest uses a new SOF_IPC_TPLG_VFE_GET VirtIO-specific
+IPC message to request the topology from the host. The host reads the file,
+corresponding to the guest ID, obtained from the QEMU communication described
+above, from its filesystem, and sends it in multiple chunks to the guest in
+reply to that request.
+
+Guest topology files contain virtual DAI components of type "dai_out" and
+"dai_in." Their data tuples contain a new token of type SOF_TKN_COMP_REF, which
+contains an ID of the connection point, to which this pipeline should be
+attached.
+
+After obtaining the topology, the guest SOF instance uses a second new
+SOF_IPC_TPLG_VFE_COMP_ID VirtIO-specific IPC message to request its allocated
+component ID base. The host maintains a list of guest component ID ranges to
+guarantee, that component IDs stay unique regardless of the order in which
+guests are brought up and down.
```

Comment:
can you clarify this: around Line 178 you write "volume component ID as seen by the respective guest. Therefore it should be exactly the same ID, as the one used by the firmware."

so how does the notion of guess-specific component ID base work?